src/rgw/rgw_bucket_logging.cc

```diff
@@ -0,0 +1,476 @@
+    return -EINVAL;
+  }
+  std::unique_ptr<rgw::sal::Bucket> target_bucket;
+  auto ret = driver->load_bucket(dpp, rgw_bucket(s->bucket_tenant, conf.target_bucket),
+                               &target_bucket, y);
+  if (ret < 0) {
+    ldpp_dout(dpp, 1) << "ERROR: failed to get target logging bucket '" << conf.target_bucket << "'. ret = " << ret << dendl;
+    return ret;
+  }
+  std::string obj_name;
+  RGWObjVersionTracker objv_tracker;
+  ret = target_bucket->get_logging_object_name(obj_name, conf.target_prefix, y, dpp, &objv_tracker);
+  if (ret == 0) {
+    const auto time_to_commit = time_from_name(obj_name, dpp) + std::chrono::seconds(conf.obj_roll_time);
+    if (ceph::coarse_real_time::clock::now() > time_to_commit) {
+      ldpp_dout(dpp, 20) << "INFO: logging object '" << obj_name << "' exceeded its time, will be committed to bucket '" <<
+        conf.target_bucket << "'" << dendl;
+      if (ret = rollover_logging_object(conf, target_bucket, obj_name, dpp, y, false, &objv_tracker); ret < 0) {
+        return ret;
+      }
+    } else {
+      ldpp_dout(dpp, 20) << "INFO: record will be written to current logging object '" << obj_name << "'. will be comitted at: " << time_to_commit << dendl;
+    }
+  } else if (ret == -ENOENT) {
+    // try to create the temporary log object for the first time
+    ret = new_logging_object(conf, target_bucket, obj_name, dpp, y, true, nullptr);
+    if (ret == 0) {
+      ldpp_dout(dpp, 20) << "INFO: first time logging for bucket '" << conf.target_bucket << "'" << dendl;
+    } else if (ret == -ECANCELED) {
+      ldpp_dout(dpp, 20) << "INFO: logging object '" << obj_name << "' already exists for bucket '" << conf.target_bucket << "', will be used" << dendl;
+    } else {
+      ldpp_dout(dpp, 1) << "ERROR: failed to create logging object of bucket '" <<
+        conf.target_bucket << "' for the first time. ret = " << ret << dendl;
+      return ret;
+    }
+  } else {
+    ldpp_dout(dpp, 1) << "ERROR: failed to get name of logging object of bucket '" <<
+      conf.target_bucket << "'. ret = " << ret << dendl;
+    return ret;
+  }
+
+  std::string record;
+  const auto tt = ceph::coarse_real_time::clock::to_time_t(s->time);
+  std::tm t{};
+  localtime_r(&tt, &t);
+  auto user_or_account = s->account_name;
+  if (user_or_account.empty()) {
+    s->user->get_id().to_str(user_or_account);
+  }
+  auto fqdn = s->info.host;
+  if (!s->info.domain.empty() && !fqdn.empty()) {
+    fqdn.append(".").append(s->info.domain);
+  }
+  switch (conf.logging_type) {
+    case LoggingType::Standard:
+      record = fmt::format("{} {} [{:%d/%b/%Y:%H:%M:%S %z}] {} {} {} REST.{}.{} {} \"{} {}{}{} HTTP/1.1\" {} {} {} {} {} {} {} {} {} {} {} {} {} {} {} {} {}",
+        dash_if_empty(to_string(s->bucket->get_owner())),
+        dash_if_empty(s->bucket->get_name()),
+        t,
+        "-", // no requester IP
+        dash_if_empty(user_or_account),
+        dash_if_empty(s->req_id),
+        s->info.method,
```

Comment:
@AliMasarweh please fix the formatting here to use the new canonical name, instead of the formatting